sphinxdocs/private/proto_to_markdown.py

```diff
@@ -0,0 +1,429 @@
+                        field.doc_string,
+                    ]
+                )
+            self._render_field_list(entries)
+
+    def _render_attributes(self, base_anchor, attributes):
+        self._write(
+            _block_attrs(f"{base_anchor}_attributes"),
+            "**ATTRIBUTES** ",
+            _link_here_icon(f"{base_anchor}_attributes"),
+            "\n",
+        )
+        entries = []
+        for attr in attributes:
+            anchor = f"{base_anchor}_{attr.name}"
+            required = "required" if attr.mandatory else "optional"
+            attr_type = self._rule_attr_type_string(attr)
+            default = f", default `{attr.default_value}`" if attr.default_value else ""
+            # The text has to be indented to be associated with the block correctly.
+            entries.append(
+                [
+                    anchor,
+                    attr.name,
+                    f"_({required} {attr_type}{default})_\n",
+                    attr.doc_string,
+                ]
+            )
+        self._render_field_list(entries)
+
+    def _render_signature(
+        self, name, base_anchor, parameters, *, get_name, get_default=lambda v: None
+    ):
+        self._write(_block_attrs(".starlark-signature"), name, "(")
+        for _, is_last, param in _position_iter(parameters):
+            param_name = get_name(param)
+            self._write(_link(param_name, f"{base_anchor}_{param_name}"))
+            default_value = get_default(param)
+            if default_value:
+                self._write(f"={default_value}")
+            if not is_last:
+                self._write(",\n")
+        self._write(")\n\n")
+
+    def _render_field_list(self, entries):
+        for anchor, description, *body_pieces in entries:
+            body_pieces = [_block_attrs(anchor), *body_pieces]
+            self._write(
+                ":",
+                _span(description + _link_here_icon(anchor)),
+                ":\n  ",
+                # The text has to be indented to be associated with the block correctly.
+                "".join(body_pieces).strip().replace("\n", "\n  "),
+                "\n",
+            )
+
+    def _write(self, *lines):
+        self._out_stream.writelines(lines)
+
+
+def _convert(*, proto_path, output_path, footer_path, public_load_path):
+    if footer_path:
+        footer_content = pathlib.Path(footer_path).read_text()
+
+    module = stardoc_output_pb2.ModuleInfo.FromString(
+        pathlib.Path(proto_path).read_bytes()
+    )
+    with pathlib.Path(output_path).open("wt", encoding="utf8") as out_stream:
+        MySTRenderer(module, out_stream, public_load_path).render()
+        out_stream.write(footer_content)
+
+
+def _create_parser():
+    parser = argparse.ArgumentParser(fromfile_prefix_chars="@")
+    parser.add_argument("--footer", dest="footer")
+    parser.add_argument("--proto", dest="proto")
+    parser.add_argument("--output", dest="output")
```

Comment:
having these as Paths may be nicer in other functions.

```suggestion
    parser.add_argument("--footer", dest="footer", type=pathlib.Path)
    parser.add_argument("--proto", dest="proto", type=pathlib.Path)
    parser.add_argument("--output", dest="output", type=pathlib.Path)
```